### ISSUE TITLE
std::bad_alloc error

### ISSUE THREAD

**Issue by DBS-Max** [luntergroup/octopus#97]
**Describe the bug**
2019-12-06 16:40:19] <DEBG> Merging 3366 temporary VCF files
[2019-12-06 16:41:40] <INFO> Starting Call Set Refinement (CSR) filtering
[2019-12-06 16:41:46] <DEBG> Encountered an error whilst filtering, attempting to cleanup
[2019-12-06 16:41:46] <INFO> Removed 6209 temporary files
[2019-12-06 16:41:47] <EROR> A program error has occurred:
[2019-12-06 16:41:47] <EROR> 
[2019-12-06 16:41:47] <EROR>     Encountered an exception during calling 'std::bad_alloc'. This means
[2019-12-06 16:41:47] <EROR>     there is a bug and your results are untrustworthy.
[2019-12-06 16:41:47] <EROR> 
[2019-12-06 16:41:47] <EROR> To help resolve this error run in debug mode and send the log file to
[2019-12-06 16:41:47] <EROR> https://github.com/luntergroup/octopus/issues.
[2019-12-06 16:41:47] <INFO> ------------------------------------------------------------------------

**Command**
Command line to run octopus:
```shell
octopus \
      -R Homo_sapiens_assembly38.fasta \
      -I 0_NA12878.bam \
      --sequence-error-model PCR-FREE.NOVASEQ \
      --forest ~/octopus/resources/forests/germline.v0.7.0.forest \
      -o 0_NA12878.bcf \
      --threads 32 \
      --debug
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04.3 LTS (Bionic Beaver)
 - Version 0.7.0
 - Reference GRCh38

**Additional context**
I have a 3.8GB debug log file, not sure how to get it to you though

**Comment by DBS-Max** [comment]
I am running octopus on ~1x bam files, I'm not sure if that makes a difference

**Comment by dancooke** [comment]
Please could you provide the exact Octopus version you're using (or just post output from `octopus --version` command)?

**Comment by DBS-Max** [comment]
I am not sure which version I was using at the time, i have since rebuilt the docker container with release branch 0.7.0 and the error has so far resolved itself. I will let you know if it happens again.

However, multithreading doesn't seem to be working very well on this branch, it barely breaks about 100% cpu usage ever with 4 threads given.

Finally, I can't download the somatic forest though the genomic one makes it fine.

**Comment by dancooke** [comment]
Ok, I'll close this issue then. Please reopen if the problem reoccurs.

> However, multithreading doesn't seem to be working very well on this branch, it barely breaks about 100% cpu usage ever with 4 threads given.

Maybe because the main bottleneck is file IO which is limited to a single thread. Are you getting steady calling progress?

> Finally, I can't download the somatic forest though the genomic one makes it fine.

The v0.7.0 forest on Google cloud was experimental and will not work with the upcoming v0.7.0 release (I've removed in from Google cloud now). Germline and somatic forests for v0.7.0 will be released when the v0.7.0 branch is merged into master and tagged - the v0.7.0 release branch is still active.

**Comment by tzvi909** [comment]
this needs to be reopened. same problem occuring on v.0.7.4
